src/ajax.js

```diff
@@ -66,13 +69,18 @@ function getCORS(url, callback) {
 }
 
 // ----------
-function getJSONP(url, callback) {
+function getJSONP(url, callback, failure) {
   var script = document.createElement('script');
   script.async = true;
   var id = '_jsonp' + sequence++;
+  var aborted = false;
 
   var timeout = setTimeout(function() {
     util.error('Timeout trying to retrieve ' + url);
+    if (failure) {
+      failure();
+      aborted = true;
```

Comment:
Do we always want to set aborted=true?

Reply:
I believe so... That's how we make sure you only get called back once. Otherwise you might get a `failure` and then later a `callback` (i.e. success) for the same call, which is a more difficult API to use (not knowing how many callbacks you'll get).

Reply:
In this code, `aborted` is only set to `true` if `failure` is defined. Should we set `aborted` to `true` even if `failure` is undefined?